### PR TITLE
Linux port

### PR DESCRIPTION
**Pull request by netanelgonen**

### CODE REVIEW

**Comment by yogpan01** [comment]
@netanelgonen What's the logic of removing the code ? Please update this PR with only CMakeList change.

File: nanostack-event-loop/platform/eventloop_config.h [hunk 1]
```diff
@@ -19,8 +19,8 @@
 /*
  * Undefine all internal flags before evaluating the configuration
  */
-#undef NS_EVENTLOOP_USE_TICK_TIMER
-#undef NS_EXCLUDE_HIGHRES_TIMER
+//#undef NS_EVENTLOOP_USE_TICK_TIMER
```
Comment (kjbracey-arm) [inline nanostack-event-loop/platform/eventloop_config.h]: Yes, these shouldn't have been undeffing. Suggest leaving them as comment out `#define`s, and replace the "undefine" comment with a brief explanation of each option.

Maybe

TICK_TIMER: "Define to use platform_tick_timer API for low-resolution timers"
EXCLUDE_HIGH_RES_TIMER: "Define if platform does not provide high-resolution platform_timer API"

File: source/event.c [hunk 2]
```diff
@@ -190,6 +190,7 @@ void event_core_write(arm_core_event_s *event)
     bool added = false;
     ns_list_foreach(arm_core_event_s, event_tmp, &event_queue_active) {
         // note enum ordering means we're checking if event_tmp is LOWER priority than event
+
```
Comment (kjbracey-arm) [inline source/event.c]: You shouldn't be touching this file at all. Squash your commits together, and make sure this file is untouched.

File: source/event.c [hunk 3]
```diff
@@ -244,15 +245,15 @@ void eventOS_scheduler_set_active_tasklet(int8_t tasklet)
 {
     curr_tasklet = tasklet;
 }
-
+/*
```
Comment (kjbracey-arm) [inline source/event.c]: I think I've belatedly figured out why this is commented out - it doesn't link if EXCLUDE_HIGH_RES_TIMER is defined?

Yes, it looks like this needs a proper substitute implementation for the NS_EVENTLOOP_USE_TICK_TIMER case. Or is it a dummy in that case? Maybe @kuggenhoffen would like to check.

**Comment by netanelgonen** [comment]
decided ti handle port from higher level

**Comment by kjbracey-arm** [comment]
So what are you doing about the high-res timer config glitch? That issue apparently still needs to be dealt with.

Is anyone planning to put in a new PR for that?

**Comment by netanelgonen** [comment]
to solve this issue we have added a stub function.

currently we are not going to handle this isuue from our side.
id needed i can open a defect

**Comment by kjbracey-arm** [comment]
Stub function for that is a reasonable workaround.

I'm a bit more concerned about the config though. I guess you're dealing with that by defining `MBED_CONF` options rather than `NS_EVENTLOOP` options? Bit icky.

I'd rather see this fixed properly here, but as it appears you have non-fragile workarounds, I guess that's fine for now.

I'll add an issue to record this.

File: nanostack-event-loop/platform/eventloop_config.h (hunk 1 above)
Comment (kuggenhoffen) [inline nanostack-event-loop/platform/eventloop_config.h]: I think the idea was that if needed these would be defined by user provided and defined config file that gets included at the end of this file.

Comment (kjbracey-arm) [inline nanostack-event-loop/platform/eventloop_config.h]: Aha! Never spotted the user config file. That's not the way any of the Nanomesh components have worked before, so wasn't expecting it.

(Usual convention is options can be passed in from Make, or picked up by known config systems like mbed or Yotta JSON).

Well, that works - I just think I'll expand the comments above the ifdefs to make it clearer.

**Pull request closed without merging**